core/src/main/java/org/hibernate/ogm/storedprocedure/impl/NoSQLProcedureCallImpl.java

```diff
@@ -0,0 +1,133 @@
+import org.hibernate.internal.util.StringHelper;
+import org.hibernate.ogm.dialect.storedprocedure.spi.StoredProcedureAwareGridDialect;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+import org.hibernate.procedure.ParameterRegistration;
+import org.hibernate.procedure.ProcedureOutputs;
+import org.hibernate.procedure.internal.ProcedureCallImpl;
+import org.hibernate.procedure.spi.ParameterStrategy;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class NoSQLProcedureCallImpl extends ProcedureCallImpl {
+	private static final Log log = LoggerFactory.make();
+	private final boolean globalParameterPassNullsSetting = true;
+
+	private ParameterStrategy parameterStrategy = ParameterStrategy.UNKNOWN;
+	private List<NoSQLProcedureParameterRegistration<?>> registeredParameters = new ArrayList<NoSQLProcedureParameterRegistration<?>>();
+	private StoredProcedureAwareGridDialect gridDialect;
+
+	public NoSQLProcedureCallImpl(SessionImplementor session, String procedureName) {
+		super( session, procedureName );
+		this.gridDialect = getGridDialectService();
+		defineParameterStrategy();
+	}
+
+	public NoSQLProcedureCallImpl(SessionImplementor session, String procedureName, Class[] resultClasses) {
+		super( session, procedureName, resultClasses );
+		this.gridDialect = getGridDialectService();
+		defineParameterStrategy();
+	}
+
+	public NoSQLProcedureCallImpl(SessionImplementor session, String procedureName, String... resultSetMappings) {
+		super( session, procedureName, resultSetMappings );
+		this.gridDialect = getGridDialectService();
+	}
+
+	private void defineParameterStrategy() {
+		if ( this.gridDialect.supportsNamedPosition() ) {
+			parameterStrategy = ParameterStrategy.NAMED;
+		}
+		else {
+			parameterStrategy = ParameterStrategy.POSITIONAL;
+		}
+	}
+
+	@Override
+	public List<ParameterRegistration> getRegisteredParameters() {
+		List<ParameterRegistration> resList = new ArrayList<>( registeredParameters.size() );
+		for ( NoSQLProcedureParameterRegistration reg : registeredParameters ) {
+			resList.add( reg );
+		}
+		return resList;
+	}
+
+	@Override
+	public <T> ParameterRegistration<T> registerParameter(String name, Class<T> type, ParameterMode mode) {
+		NoSQLProcedureParameterRegistration<T> parameterRegistration = new
+				NoSQLProcedureParameterRegistration<T>( this, null, name, mode, type );
+		registerParameter( parameterRegistration );
+		return parameterRegistration;
+	}
+
+	@Override
+	public <T> ParameterRegistration<T> registerParameter(int position, Class<T> type, ParameterMode mode) {
+		NoSQLProcedureParameterRegistration<T> parameterRegistration =
+				new NoSQLProcedureParameterRegistration<T>( this, position, null, mode, type );
+		registerParameter( parameterRegistration );
+		return parameterRegistration;
+	}
+
+	@Override
+	public ProcedureOutputs getOutputs() {
+		log.info( "I am here!" );
```

Comment:
Terrible log

Reply:
yes. I needed these logs for make travel in lot of classes :). Now ... logs like this  must be removed. I will do it.

Reply:
I usually work with a debugger